cmd/kubectl/kubectlcobra/grouping.go

```diff
@@ -211,3 +219,31 @@ func mapKeysToSlice(m map[string]string) []string {
 	}
 	return s
 }
+
+// addSuffixToName adds the passed suffix (usually a hash) as a suffix
+// to the name of the passed object stored in the Info struct. Returns
+// an error if the object is not "*unstructured.Unstructured" or if the
+// name stored in the object differs from the name in the Info struct.
+func addSuffixToName(info *resource.Info, suffix string) error {
+
+	if info == nil {
+		return fmt.Errorf("Nil resource.Info")
+	}
+	if len(suffix) == 0 {
+		return nil
+	}
+
+	accessor, _ := meta.Accessor(info.Object)
+	name := accessor.GetName()
+	if name != info.Name {
+		return fmt.Errorf("Grouping object (%s) and resource.Info (%s) have different names\n", name, info.Name)
+	}
+	// Only add the suffix if it has not already been added.
```

Comment:
i think we should add the hyphen to the suffix here (suffix = "-" + suffix) then check to see if its already there and if so, return an error (since that's likely something we've screwed up higher up).